crates/host/src/wasmbus/mod.rs

```diff
@@ -2164,16 +2175,55 @@ impl Host {
         Ok(buf.into())
     }
 
-    #[allow(unused)] // TODO: Remove once implemented
-    #[instrument(skip(self, payload))]
-    async fn handle_claims(&self, payload: impl AsRef<[u8]>) -> anyhow::Result<Bytes> {
-        bail!("TODO")
+    #[instrument(skip(self))]
+    async fn handle_claims(&self) -> anyhow::Result<Bytes> {
+        // TODO: update control interface client to have a more specific type definition for
+        // GetClaimsResponse, so we can re-use it here. Currently it's Vec<HashMap<String, String>>
+        #[derive(Serialize)]
+        struct ClaimsResponse {
+            claims: Vec<StoredClaims>,
+        }
+
+        let claims: Vec<StoredClaims> = self.scan_latticedata(CLAIMS_PREFIX).await?;
+        let resp = ClaimsResponse { claims };
+        Ok(serde_json::to_vec(&resp)?.into())
     }
 
-    #[allow(unused)] // TODO: Remove once implemented
-    #[instrument(skip(self, payload))]
-    async fn handle_links(&self, payload: impl AsRef<[u8]>) -> anyhow::Result<Bytes> {
-        bail!("TODO")
+    #[instrument(skip(self))]
+    async fn handle_links(&self) -> anyhow::Result<Bytes> {
+        let links: Vec<LinkDefinition> = self.scan_latticedata(LINKDEF_PREFIX).await?;
+        let resp = LinkDefinitionList { links };
+        Ok(serde_json::to_vec(&resp)?.into())
+    }
+
+    async fn scan_latticedata<T: DeserializeOwned>(
+        &self,
+        key_prefix: &str,
+    ) -> anyhow::Result<Vec<T>> {
+        let filtered_keys = self
+            .data
+            .keys()
+            .await
+            .context("failed to scan lattice data keys")?
+            .try_filter(|key| futures::future::ready(key.starts_with(key_prefix)))
+            .try_collect::<Vec<String>>()
+            .await
+            .context("failed to collect lattice data keys")?;
+
+        let futs = filtered_keys.into_iter().map(|key| self.data.get(key));
```

Comment:
`try_collect` here is redundant, since immediately after allocating a whole vector you're turning it into an iterator and then allocating yet another vector of futures.

Try something like this:
```suggestion
           .and_then(|key| async {
              let buf = self.data.get(key).await.context("failed to get lattice key")?;
              serde_json::from_slice::<T>(&buf)
           });
```

Reply:
This approach resolves https://github.com/wasmCloud/wasmCloud/issues/509, but I'm not sure it's the best behavior: if we fail to retrieve any value — say a key is deleted while processing, or there's a malformed entry in the bucket — do we want to fail to return anything to the client? Wouldn't it be better to exclude bad values and emit a warning?

Reply:
Yeah, I think we should print an `error` log if we fail to deserialize an entry here, but I'm not suggesting any change in behavior here, just removing the redundant steps

Reply:
I was confused by your code snippet, where failing to deserialize would also result in a top-level error. I've refactored the code and added comments clarifying which errors I think we should gracefully bypass